src/styles/transformers/rtl.js

```diff
@@ -0,0 +1,90 @@
+import merge from '../../merge';
+import warning from 'warning';
+
+const reTranslate = /((^|\s)translate(3d|X)?\()(\-?[\d]+)/;
+const reSkew = /((^|\s)skew(x|y)?\()\s*(\-?[\d]+)(deg|rad|grad)(,\s*(\-?[\d]+)(deg|rad|grad))?/;
+
+/**
+ * This function ensures that `style` supports both ltr and rtl directions by
+ * checking `styleConstants` in `muiTheme` and replacing attribute keys if
+ * necessary.
+ */
+export default function rtl(muiTheme, style) {
```

Comment:
how about: 

```js
export default theme => style => {
//...
};
```

This will be like a transformer factory function and it will dramatically simplify your `createPrepareStyles` function, which took me some time to understand :grin:. since you won't need to wrap before chain complication the behavior.